beacon_node/beacon_chain/src/validator_monitor.rs

```diff
@@ -489,6 +509,131 @@ impl<T: EthSpec> ValidatorMonitor<T> {
                 }
             }
         }
+
+        // Add missed non-finalized blocks for the monitored validators
+        // count the amount of times a monitored validator missed a non-finalized block
+        self.missed_blocks
+            .iter()
+            .group_by(|(_, validator_index, _)| *validator_index)
+            .into_iter()
+            .for_each(|(validator_index, group)| {
+                let missed_blocks_count = group.count();
+                self.aggregatable_metric(validator_index.to_string().as_str(), | label| {
+                    metrics::set_int_gauge(
+                        &metrics::VALIDATOR_MONITOR_MISSED_NON_FINALIZED_BLOCKS_AT_NON_FINALIZED_EPOCH_TOTAL,
+                        &[label],
+                        u64_to_i64(missed_blocks_count as u64),
+                    );
+                });
+            });
+
+        // add to the total missed blocks by monitored validator
+        self.total_missed_blocks
```

Comment:
The `to_owned` here is cloning the `total_missed_blocks` hash map, which costs heap allocations.

We can avoid this be replacing the `.to_owned().into_iter()` with `.iter()` as long as we dereference `c` (e.g. `*c as i64`).